orttraining/orttraining/core/framework/gradient_graph_builder.cc

```diff
@@ -51,19 +54,25 @@ GradientGraphBuilder::GradientGraphBuilder(Graph* graph,
       }
       ORT_THROW("Node arg '", name, "' is not found in the graph. Available output names = ", ss.str());
     }
-    y_node_args_.insert(node_arg);
 
     const Node* node = graph_->GetProducerNode(name);
     if (!node) {
       ORT_THROW(name, " couldn't find the producer node.");
     }
-    y_nodes_.insert(node);
+
+    if (forward_reachable_nodes.find(node) == forward_reachable_nodes.end()) {
+      non_differentiable_y_node_arg_names_.insert(name);
+      LOGS(logger_, WARNING) << "The model weigths and inputs are non-differentiable from " << name << ". "
```

Comment:
```suggestion
      LOGS(logger_, INFO) << "The model weights and inputs are non-differentiable from " << name << ". "
```
Should it be INFO? We are trying to suppress all messages that the user cannot do anything about it during training. Currently implementation is verbose

nit: typo

Reply:
done.